baiji/s3.py

```diff
@@ -839,8 +847,8 @@ def exists(self, key_or_file, retries_allowed=3):
         elif k.scheme == 's3':
             retry_attempts = 0
             while retry_attempts < retries_allowed:
-                key_exists = self._lookup(k.netloc, k.path, cache_buckets=True) is not None
-                if key_exists:
+                key = self._lookup(k.netloc, k.path, cache_buckets=True)
```

Comment:
I was mistakenly returning `False` instead of `None` from `_lookup` because I didn't realize what I should be returning. That made `s3.exists` return the wrong value.

This change seems a bit more Pythonic and slightly safer.